app/models/SignOutReason.scala

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models
+
+import play.api.mvc.QueryStringBindable
+
+sealed abstract class SignOutReason(val name: String)
```

Comment:
I am always amazed by what a pain "enums" are in Scala and the hoops you have to jump thought to serialise them.

What's wrong with a nice String?  (Not suggesting you change it - just having a rant)

Reply:
Strings are too generic in my opinion. Prefer to have types if only possible. Using type here limits allowed values and also allows for 'cleaner' pattern matching in controller.
Besides, it requires more code not because of being "enum" but because this type is used as URL query parameter.

Reply:
Yes, I know the benefits of enums.  I was just commenting that they are a pain to use in our services.  I'm fairly sure that using Java and SpringBoot you could just define the enum in a couple of lines and use it as a url param without any of this bind/unbind malarky.